BibBuddy/app/src/main/java/de/bibbuddy/HomeFragment.java

```diff
@@ -1,38 +1,151 @@
 package de.bibbuddy;
 
+import android.content.Context;
 import android.os.Bundle;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.TextView;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
-import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.RecyclerView;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
 
 /**
- * The HomeFragment is responsible for TODO.
+ * The HomeFragment is responsible for welcoming the user and displaying recently used books.
  *
- * @author Claudia Schönherr
+ * @author Claudia Schönherr, Luis Moßburger
  */
-public class HomeFragment extends Fragment {
+public class HomeFragment extends BackStackFragment
+    implements BookRecyclerViewAdapter.BookListener {
 
   private View view;
+  private Context context;
+
+  private BookDao bookDao;
+  private BookModel bookModel;
+  private DatabaseHelper dbHelper;
+
+  private int bookAmount;
+  List<BookItem> bookItemList;
+  private BookRecyclerViewAdapter adapter;
 
   @Nullable
   @Override
   public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                            @Nullable Bundle savedInstanceState) {
 
     view = inflater.inflate(R.layout.fragment_home, container, false);
+    context = view.getContext();
```

Comment:
```suggestion
    context = view.requireContext();
```
Replace the getContext() with requireContext() to avoid NullPointerExceptions.

Reply:
As far as I can see, the correct implementation would be the following:

`
    context = requireContext();
`